boot/bootutil/src/loader.c

```diff
@@ -2488,6 +2521,11 @@ context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
                 rc = boot_perform_update(state, &bs);
             }
             assert(rc == 0);
+            if (is_owner_nsib(state)) {
+#ifdef __ZEPHYR__
```

Comment:
Since this is a downstream patch is this strictly needed? or could we replace it with NVIC_SystemReset() if we are worried that someone is building our downstream version of MCUBoot with another RTOS?

Reply:
`sys_reboot` is fine, I don't think anyone would be using this for non-zephyr builds because this is part of ncs specifically and we don't support other rtos's with this code or any of the ncs libraries

Reply:
Though the one thing I would want checked with this before having this merged is that this is actually ok for customers, because I have no idea what they are using b0 + mcuboot updates for, if it is just for mcuboot bug fixes then this is fine, if it is for being able to change the mcuboot key then this will break booting any applications unless they have serial recovery enabled - but I don't know who would have such information. Sales? FAEs?

Reply:
> `sys_reboot` is fine, I don't think anyone would be using this for non-zephyr builds because this is part of ncs specifically and we don't support other rtos's with this code or any of the ncs libraries

Then we could drop the `#ifdef`.

> If they are expecting the old image to boot the first time after updating mcuboot so they can then load a new application update (this is a very dodgy thing because a single power failure or reboot before the new image is loaded will brick the device) then there might need to be a Kconfig to control this

This is a good point so having a KConfig toogle makes sense as maybe some customers who gets their keys compromised would actually retain this feature of not rebooting.

Reply:
I don't know about such a risky cascade update method. I can imagine that rather mcuboot would be update  to a (non existing jet) version with key revocation feature.